### ISSUE TITLE
fail to update bosh-link configuration on pod-0 when pod-1 is restarted

### ISSUE THREAD

**Issue by cdlliuy** [cloudfoundry-incubator/configgin#90]
To enable sharding mechanism ,  we used bosh.link feature to get each node information and write the node's ip address and port into configuration.yml. 
```
  sorted_instances=link("metricscollector").instances.sort_by {|i|i.address}
  nodeIndex=sorted_instances.index(sorted_instances.find{|i|i.id == spec.id})
  addrs=sorted_instances.map{|i| "'#{i.address}:#{p('autoscaler.metricscollector.server.port')}'"}
  nodeAddrs="[" + addrs.join(",") +"]"  
```
Then, the resulted configuration looks like:
```
  node_addrs: ['172.30.147.84:7103','172.30.206.99:7103']
  node_index: 0
```
The bosh-link render works good when creating multiple pod. But ... when one pod is restarted,  it will gain a new IP address. In this situation,  the other un-changed pod will get the configuration refresh, then the node_addrs in the configuration is wrong ...

See below example: 

At the very beginning:
```
NAME                            READY STATUS      RESTARTS   AGE       IP               NODE
autoscaler-metrics-0   2/2       Running   0         4s        172.30.147.88   10.73.120.54
autoscaler-metrics-1   2/2       Running   0         12s       172.30.206.101   10.73.120.28
```
The configuration in `autoscaler-metrics-0` is: 
```
  node_addrs: ['172.30.147.88:7103','172.30.206.101:7103']
  node_index: 0
```
The configuration in `autoscaler-metrics-1` is: 
```
  node_addrs: ['172.30.147.88:7103','172.30.206.101:7103']
  node_index: 1
```
Up to now, everything is good. 

Then, I deleted autoscaler-metrics-0 , and k8s created it again. 
Now, 
autoscaler-metrics-0   2/2       Running   0         4s        172.30.147.84   10.73.120.54

The configuration in `autoscaler-metrics-0` is correct: 
```
  node_addrs: ['172.30.147.84:7103','172.30.206.101:7103']
  node_index: 0
```
But the configuration in `autoscaler-metrics-1` still is: 
```
  node_addrs: ['172.30.147.88:7103','172.30.206.101:7103']
  node_index: 1
```
The ip address 172.30.147.88 is totally out-of-date now.  

Is that possible to fix this link issue?

**Comment by cdlliuy** [comment]
the same scenario is not a problem when deploying on bosh-lite vm, since restarting a vm through bosh won't change its IP address ..

**Comment by cdlliuy** [comment]
any update on this?  
Is that possible to use dns name instead of ip address here?  According to "https://bosh.io/docs/links/", it is possible to be a DNS address.
```
address [String, non-empty]: IPv4, IPv6 or DNS address. See Native DNS Support for more details.
```

**Comment by zhanggbj** [comment]
With current logic in configgin, it only supports **pod IP address** for field `address ` as below, it does not support DNS address as @cdlliuy pasted above. 
```
  def get_pod_instance_info(pod, job, pods_per_image)
    index = pod_index(pod.metadata.name)
    {
      'name' => pod.metadata.name,
      'index' => index,
      'id' => pod.metadata.name,
      'az' => pod.metadata.annotations['failure-domain.beta.kubernetes.io/zone'] || 'az0',
      'address' => pod.status.podIP,
      'properties' => get_exported_properties(pod, job),
      'bootstrap' => pods_per_image[pod.metadata.uid] < 2
    }
```

And I think on K8s, the **DNS address** is a more proper value for field `address`, as everytime a pod restart with a **new** IP address, which is not the same behavior as BOSH, configgin can change to use DNS address like `podname.servicename.namespace.svc.cluster.local` instead of IP address.

**Comment by zhanggbj** [comment]
Hi @mook-as, 
I saw that PR #93 has been merged. Thank you. And one more question, will this commit be included in next version of configgin gem? Is there any expected date? As autoscaler team has one PR pending on this. Thanks!

https://rubygems.org/gems/configgin/versions

**Comment by mook-as** [comment]
@zhanggbj Ah! I didn't realize you were in a hurry on this one; sorry about that. Please make a PR to [bump the version](https://github.com/cloudfoundry-incubator/configgin/blob/develop/lib/configgin/version.rb) (probably minor for this one, as it shouldn't be a breaking change), and I'll do a release when merging, thanks!

**Comment by zhanggbj** [comment]
@mook-as 
Great! Thanks a lot for your help :-) I've raised a new PR to bump version. Thanks！

**Comment by jandubois** [comment]
I've released the updated gem: https://rubygems.org/gems/configgin/versions/0.18.3
